contrib/proguard/src/Proguard.scala

```diff
@@ -0,0 +1,82 @@
+package mill.contrib.proguard
+
+import coursier.Repositories
+import mill.T
+import mill.Agg
+import mill.api.{Logger, Loose, PathRef, Result}
+import mill.define.{Sources, Target}
+import mill.scalalib.Lib.resolveDependencies
+import mill.scalalib.{Dep, DepSyntax, Lib, ScalaModule}
+import os.proc
+import os.Path
+import os.PathChunk
+
+trait Proguard extends ScalaModule {
+
+  // This is actually the only available version on Maven currently.
+  // But we should keep it as a configuration in case a new version is released.
+  def proguardVersion: T[String] = T { "7.0.0" }
+
+  def shrink: T[Boolean] = T { true }
+  def optimize: T[Boolean] = T { true }
+  def obfuscate: T[Boolean] = T { true }
+  def preverify: T[Boolean] = T { true }
+
+  def javaHome: T[PathRef] = T {
+    PathRef(Path(System.getProperty("java.home")))
+  }
+
+  def inJar: T[PathRef] = T { assembly() }
+  def outJar: T[PathRef] = T { PathRef(T.dest / "out.jar") }
```

Comment:
Why is this needed. Isn't it always the output of the `proguard` target?

Reply:
Yeah you're right